add_codeowners.py

```diff
@@ -0,0 +1,498 @@
+    commit_message: str = "MNT: Adding CODEOWNERS file",
+    title="MNT: Adding CODEOWNERS file",
+    body="(autogenerated) adding codeowners file",
+    sme: Optional[List[str]] = None,
+    area: Optional[List[str]] = None,
+    lang: Optional[List[str]] = None,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    """
+    Create codeowners files and submit them via pull request
+    """
+    # Set dry-run option
+    DryRunner.set(run=write)
+    api = GhApi()
+
+    # parse csv with settings
+    if repo_data_path:
+        repo_data = parse_repo_list(repo_data_path)
+        for settings in repo_data:
+            print(f"--- working on {settings.repo_name}...")
+            add_codeowners_from_setting(
+                settings, user, branch_name,
+                commit_message, title, body,
+                api=api
+            )
+
+        return
+
+    if not repo_name:
+        print("Neither repo name nor repo data path provided, nothing to do")
+        return
+
+    # apply on a per-repo basis
+    sme_owners = [CodeownerGroup(grp) for grp in sme]
+    area_owners = [CodeownerGroup(grp) for grp in area]
+    lang_owners = [CodeownerGroup(grp) for grp in lang]
+
+    print(sme_owners, area_owners, lang_owners)
+
+    settings = RepoOwnerSettings(
+        owner=owner,
+        repo_name=repo_name,
+        sme_owners=sme_owners,
+        area_owners=area_owners,
+        lang_owners=lang_owners,
+    )
+
+    print(f"--- working on {settings.repo_name}...")
+    add_codeowners_from_setting(
+        settings, user, branch_name,
+        commit_message, title, body,
+        api=api
+    )
+
+
+def _create_argparser() -> argparse.ArgumentParser:
+    """
+    Create an ArgumentParser for add_codeowners
+
+    If we ever reuse this we should probably let users specify their commit
+    message, pr title, etc.  But this is already an incredibly long cli call.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    # specify an owner, repo, and settings
+    parser.add_argument("owner", type=str, default='pcdshub', nargs='?',
+                        help='Organization or owner of the repository, "pcdshub"'
+                             'by default')
+    parser.add_argument("user", type=str, default='pcdshub', nargs='?',
+                        help="Name of account to contribute from, the fork will "
+                             "created on this github account if required.")
+    parser.add_argument("branch_name", type=str, default="mnt_add_codeowners", nargs='?',
+                        help="Name of branch to create on fork and submit upstream")
+
+    # manually specify settings
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("--sme", type=str, nargs='*',
+                        choices=[g.value for g in CodeownerGroup
```

Comment:
Nitpick/question: Is there any value in splitting up this enum into multiple enums rather than splitting it manually in various places?

Reply:
You're right, if I'm treating them differently I should probably properly make them different enums